### DISCUSSION TITLE
Annotations for Serialization and Deserialization

### DISCUSSION THREAD

**Discussion by tsanton** [Gremlinq/ExRam.Gremlinq#601] [Q&A]
Hi @danielcweber and thank you for this wonderful software :)

To me it looks like you're utilizing Newtonsoft to do json-serialization/deserialization. Will this stick now with System.Text.Json?
Further, if Newtonsoft is the case, I'm assuming that decorating properties with '[JsonProperty("myCamelCase")]' will sort db-side variable naming? If so -> are there any attributes that can be utilized to handle vertex labeling? The desired outcome here is that, for instance, a PersonV vertex gets labeled "person" db-side.

Looking forward to hearing back from you!

/T

**Comment by danielcweber** [comment]
Any Newtonsoft-annotations will not work.

Have a look at this usage of [UseLowerCaseLabels](https://github.com/ExRam/ExRam.Gremlinq/blob/50b4bfb94342da5078c52b08d6c48d566e80e13c/test/ExRam.Gremlinq.Core.Tests/GraphModelTest.cs#L112) and [how it's implemented](https://github.com/ExRam/ExRam.Gremlinq/blob/81e2f4357410f1e75571009ded8274fa4a2b9703/src/ExRam.Gremlinq.Core/Models/GraphElementModel.cs#L78) through `ConfigureLabels`. You should be able to go from there and make things work for you.
